test/Faker/Provider/BaseTest.php

```diff
@@ -565,6 +566,31 @@ public function testRandomElements()
         $this->assertCount(3, $allowDuplicates);
         $this->assertContainsOnly('string', $allowDuplicates);
     }
+
+    /**
+     * @dataProvider testRegexifyDataProvider
+     *
+     * @param $expectedRegex
+     * @param $input
+     */
+    public function testRegexify($expectedRegex, $input)
+    {
+        $this->assertRegExp($expectedRegex, BaseProvider::regexify($input));
+    }
+
+    public function testRegexifyDataProvider()
```

Comment:
I only covered the new behavior.
Since the `regexify` is super complex, we should think about splitting the single steps of regexify into seperate methods of some sort. This way we can provide more tests for each step in there.

Reply:
Perhaps we should add some tests for the existing behavior before we merge this in, what do you think?

Reply:
Will try and figure some out according to the comments in the function itself. But these will merely scratch a good coverage.

Reply:
Alright, added as much tests as possible. Problem is the complexity of the method itself. But this should cover a good portion of what is happening inside. Now I hope we can refactor this method with more test coverage to not break things ;)